src/test/java/org/mockito/StaticMockingExperimentTest.java

```diff
@@ -0,0 +1,223 @@
+package org.mockito;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.exceptions.verification.NoInteractionsWanted;
+import org.mockito.exceptions.verification.WantedButNotInvoked;
+import org.mockito.exceptions.verification.junit.ArgumentsAreDifferent;
+import org.mockito.invocation.Invocation;
+import org.mockito.invocation.MockHandler;
+import org.mockitoutil.TestBase;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.concurrent.Callable;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+/**
+ * This test is an experimental use of Mockito API to simulate static mocking.
+ * Other frameworks can use it to build good support for static mocking.
+ * Keep in mind that clean code never needs to mock static methods.
+ * This test is a documentation how it can be done using current public API of Mockito.
+ * This test is not only an experiment it also provides coverage for
+ * some of the advanced public API exposed for framework integrators.
+ * <p>
+ * For more rationale behind this experimental test
+ * <a href="https://www.linkedin.com/pulse/mockito-vs-powermock-opinionated-dogmatic-static-mocking-faber">see the article</a>.
+ */
+public class StaticMockingExperimentTest extends TestBase {
+
+    Foo mock = Mockito.mock(Foo.class);
+    MockHandler handler = Mockito.mockingDetails(mock).getMockHandler();
+    Method staticMethod;
+    Callable realMethod = new Callable() {
+        @Override
+        public Object call() throws Exception {
+            return null;
+        }
+    };
+
+    @Before public void before() throws Throwable {
+        staticMethod = Foo.class.getDeclaredMethod("staticMethod", String.class);
+    }
+
+    @Test
+    public void verify_static_method() throws Throwable {
+        //register staticMethod call on mock
+        Invocation invocation = Mockito.framework().getInvocationFactory().createInvocation(mock, withSettings().build(Foo.class), staticMethod, realMethod,
+            "some arg");
+        handler.handle(invocation);
+
+        //verify staticMethod on mock
+        verify(mock);
```

Comment:
I think this should call the staticMethod? The same for the next verify

Reply:
Thanks for feedback! I have added code comment to this test method explaining what and why it is implemented this way. Is it clearer now?

Reply:
Here's the comment: https://github.com/mockito/mockito/pull/1121/commits/5be550bba197d76697d69a84d9c46f92e99088dd